core/src/main/kotlin/org/evomaster/core/problem/rest/service/SecurityRest.kt

```diff
@@ -857,7 +868,179 @@ class SecurityRest {
         }
     }
 
+    /**
+     * XSS (Cross-Site Scripting) check.
+     *
+     * Creates test cases for both reflected and stored XSS by injecting malicious payloads
+     * into string parameters. For each endpoint with a 2xx response:
+     * - Injects XSS payload into string fields (body, query, path parameters)
+     * - For POST/PUT/PATCH: attempts to add a GET on the same path for stored XSS detection
+     *
+     */
+
+    private fun handleXSSCheck(){
+
+        mainloop@ for(action in actionDefinitions){
+
+
+            // Find individuals with 2xx response for this endpoint
+            val successfulIndividuals = RestIndividualSelectorUtils.findIndividuals(
+                individualsInSolution,
+                action.verb,
+                action.path,
+                statusGroup = StatusGroup.G_2xx
+            )
+
+            if(successfulIndividuals.isEmpty()){
+                continue
+            }
 
+            // Take the smallest successful individual
+            val target = successfulIndividuals.minBy { it.individual.size() }
+
+            val actionIndex = RestIndividualSelectorUtils.findIndexOfAction(
+                target,
+                action.verb,
+                action.path,
+                statusGroup = StatusGroup.G_2xx
+            )
+
+            if(actionIndex < 0){
+                continue
+            }
+
+            // Slice to keep only up to the target action
+            val sliced = RestIndividualBuilder.sliceAllCallsInIndividualAfterAction(
+                target.individual,
+                actionIndex
+            )
+
+            // Try each XSS payload (but only add one test per endpoint)
+            for(payload in XSS_PAYLOADS){
+
+                // Create a copy of the individual
+                var copy = sliced.copy() as RestIndividual
+                val actionCopy = copy.seeMainExecutableActions().last() as RestCallAction
+
+                // Try to inject XSS payload into string fields
+                var payloadInjected = false
+
+                // Attempt to inject payload into parameters
+                actionCopy.parameters.filter { it is BodyParam || it is QueryParam || it is PathParam }.forEach { param ->
```

Comment:
is this check necessary? i mean, why skipping `HeaderParam`? i guess we could skip the `filter` isn't it?